libr/include/r_anal.h

```diff
@@ -1727,33 +1717,77 @@ R_API void r_anal_data_free (RAnalData *d);
 #include <r_cons.h>
 R_API char *r_anal_data_to_string(RAnalData *d, RConsPrintablePalette *pal);
 
-R_API void r_meta_free(RAnal *m);
-R_API RList *r_meta_find_list_in(RAnal *a, ut64 at, int type, int where);
+/* meta
+ *
+ * Meta uses Condret's Klemmbaustein Priciple, i.e. intervals are defined inclusive/inclusive.
+ * A meta item from 0x42 to 0x42 has a size of 1. Items with size 0 do not exist.
+ * Meta items are allowed to overlap and the internal data structure allows for multiple meta items
+ * starting at the same address.
+ * Meta items are saved in an RIntervalTree. To access the interval of an item, use the members of RIntervalNode.
+ */
+
+static inline ut64 r_meta_item_size(ut64 start, ut64 end) {
+	// meta items use inclusive/inclusive intervals
+	return end - start + 1;
+}
+
+static inline ut64 r_meta_node_size(RIntervalNode *node) {
+	return r_meta_item_size (node->start, node->end);
+}
+
+// Set a meta item at addr with the given contents in the current space.
+// If there already exists an item with this type and space at addr (regardless of its size) it will be overwritten.
+R_API bool r_meta_set(RAnal *a, RAnalMetaType type, ut64 addr, ut64 size, const char *str);
```

Comment:
Old function was `r_meta_add`. Was it always adding a new meta?

How can i set multiple comments at the same address with this new API?

Reply:
No, `r_meta_add()` was also overwriting, despite the name. For example run this on master with #!c, then `C`:
```
#include <r_core.h>

void entry(RCore *core, int argc, const char **argv)
{
	r_meta_add(core->anal, R_META_TYPE_COMMENT, 0x100, 0x100, "comment a");
	r_meta_add(core->anal, R_META_TYPE_COMMENT, 0x100, 0x100, "comment b");
}
```

You can't set multiple comments at the same address right now (except in different spaces), which corresponds to how it was before.